presto-main/src/main/java/com/facebook/presto/operator/scalar/CharacterStringCasts.java

```diff
@@ -69,31 +71,30 @@ public static Slice varcharToCharCast(@LiteralParameter("y") Long y, @SqlType("v
     @LiteralParameters({"x", "y"})
     public static Slice charToVarcharCast(@LiteralParameter("y") Long y, @SqlType("char(x)") Slice slice)
     {
-        int textLength = countCodePoints(slice);
-        int resultLength = y.intValue();
+        return padSpacesAndTruncateToLength(slice, y.intValue());
+    }
 
-        // if our target length is the same as our string then return our string
-        if (textLength == resultLength) {
-            return slice;
-        }
+    @ScalarOperator(OperatorType.SATURATED_FLOOR_CAST)
```

Comment:
What is a saturated floor cast? I see it's implemented for a few types, but I'm not sure why we need this